### PR TITLE
Changes for SpyCat Api and ISpyBToolsApi

### PR DESCRIPTION
**Pull request by collang**

### CODE REVIEW

**Review by KarlLevik — CHANGES_REQUESTED** [review]
`.downloadschemascript.sh.swp` should be removed

File: src/main/java/uk/ac/diamond/ispyb/dao/IspybSpyCatDAO.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    @Override
+    public Optional<String> retrieveLatestVisitWithPrefix(String beamline, String proposalCode) {
+        Map<String, Object> params = new HashMap<>();
+        params.put("beamline", beamline);
+        params.put("proposal_code", proposalCode);
+        return templateWrapper.callIspyb("retrieve_most_recent_session", String.class, params);
+    }
+
+    @Override
+    public Optional<String> retrieveTitleForVisit(String proposalCode, Integer proposalNumber) {
```
Comment (KarlLevik) [inline src/main/java/uk/ac/diamond/ispyb/dao/IspybSpyCatDAO.java]: It's strictly speaking retrieving the title of the proposal, so ideally should be named accordingly.

**Review by KarlLevik — APPROVED** [review]
Looks OK, see inline comment re: method name. Also, is it too late to rename `IspybTools` to e.g. `MXCX` which is an abbreviation for "macromolecular and chemical crystallography" that seems to be favoured by developers in that area?

**Comment by collang** [comment]
I can change the name to IspybMxcxApi? That shouldn't pose a problem.

**Comment by KarlLevik** [comment]
> I can change the name to IspybMxcxApi? That shouldn't pose a problem.

Yes, perfect! (Or maybe capital C in Cx?)

**Review by KarlLevik — APPROVED** [review]
Looks good to me!

**Pull request merged**